### DISCUSSION TITLE
Detect and Refactor Duplicates (via Extract Method)

### DISCUSSION THREAD

**Discussion by dciborow** [microsoft/pylance-release#2825] [Enhancement]
Currently, I can use the "Extract Method" to take a section of code, and wrap it in a function. I should be able to detect duplicated sections of code, and easily extract them into a single common function. The duplication detection should include configurations to detect similar sections, even if minor differences exist, such as comments or different variable names. The minimum number of lines should be adjustable. These configurations should be settable via UX elements so that a developer can examine different combinations of settings quickly. I might, for example, first  want to search for duplicates that are 8 lines long, but ignore comments and variable names; and then do a different search of 3 lines but only exact matches.

**Comment by debonte** [comment]
Visual Studio has a feature called "Code Clone" that might be interesting to look at should we decide to do this.

https://visualstudiomagazine.com/articles/2011/11/17/visual-studio-11-code-clone-analysis.aspx
